Kanban/Controllers/KBNAddProjectViewController.m

```diff
@@ -53,7 +52,41 @@ - (IBAction)save:(UIBarButtonItem *)sender {
 
 - (IBAction)cancel:(UIBarButtonItem *)sender {
     [self.navigationController popToRootViewControllerAnimated:YES];
+}
 
+#pragma mark - HUD
+
+- (void)startHUD {
+    self.HUD = [[MBProgressHUD alloc] initWithView:self.view];
+    [self.view addSubview:self.HUD];
+    
+    self.HUD.dimBackground = YES;
+    self.HUD.mode = MBProgressHUDModeAnnularDeterminate;
+    
+    self.HUD.labelText = ADD_PROJECT_LOADING;
+    [self.HUD showWhileExecuting:@selector(myProgressTask) onTarget:self withObject:nil animated:YES];
+    self.HUD.delegate = self;
+}
+
+- (void)myProgressTask {
+    // This just increases the progress indicator in a loop
+    float progress = 0.0f;
+    while (progress < 1.0f) {
+        progress += 0.040f;
+        self.HUD.progress = progress;
+        usleep(50000);
+    }
+    sleep(0.7);
+    __block UIImageView *imageView;
+    dispatch_sync(dispatch_get_main_queue(), ^{
+        UIImage *image = [UIImage imageNamed:@"37x-Checkmark.png"];
+        imageView = [[UIImageView alloc] initWithImage:image];
+    });
+    self.HUD.customView = imageView;
+    self.HUD.mode = MBProgressHUDModeCustomView;
+    self.HUD.labelText = PROJECT_CREATION_SUCCESS;
+    sleep(1);
+    [self.navigationController popToRootViewControllerAnimated:YES];
```

Comment:
Is it possible to handle the navigation outside this method? For clarity, I think it should be in the "save" method.